tests/test_integrator/test_sfincs_adapter.py

```diff
@@ -1,70 +1,879 @@
-from flood_adapt.object_model.scenario import Scenario
+from flood_adapt.dbs_classes.database import Database
+from flood_adapt.dbs_classes.interface.database import IDatabase
+from flood_adapt.object_model.hazard.forcing.discharge import (
+    DischargeConstant,
+    DischargeSynthetic,
+)
+from flood_adapt.object_model.hazard.forcing.rainfall import (
+    RainfallConstant,
+    RainfallMeteo,
+    RainfallSynthetic,
+)
+from flood_adapt.object_model.hazard.forcing.waterlevels import (
+    SurgeModel,
+    TideModel,
+    WaterlevelCSV,
+    WaterlevelGauged,
+    WaterlevelModel,
+    WaterlevelSynthetic,
+)
+from flood_adapt.object_model.hazard.forcing.wind import (
+    WindConstant,
+    WindMeteo,
+    WindSynthetic,
+    WindTrack,
+)
+from flood_adapt.object_model.hazard.interface.forcing import (
+    ForcingSource,
+    ForcingType,
+    IDischarge,
+    IRainfall,
+    IWaterlevel,
+    IWind,
+)
+from flood_adapt.object_model.hazard.interface.models import (
+    TimeModel,
+)
+from flood_adapt.object_model.hazard.interface.timeseries import (
+    ShapeType,
+    SyntheticTimeseriesModel,
+)
+from flood_adapt.object_model.hazard.measure.floodwall import FloodWall
+from flood_adapt.object_model.hazard.measure.green_infrastructure import (
+    GreenInfrastructure,
+)
+from flood_adapt.object_model.hazard.measure.pump import Pump
+from flood_adapt.object_model.interface.measures import MeasureType
+from flood_adapt.object_model.interface.site import ObsPointModel, RiverModel
+from flood_adapt.object_model.io import unit_system as us
+from flood_adapt.object_model.projection import Projection
+from tests.fixtures import TEST_DATA_DIR
 
 
 @pytest.fixture()
-def test_scenarios(test_db):
-    test_tomls = [
-        test_db.input_path
-        / "scenarios"
-        / "current_extreme12ft_no_measures"
-        / "current_extreme12ft_no_measures.toml"
-    ]
-
-    test_scenarios = {
-        toml_file.name: Scenario.load_file(toml_file) for toml_file in test_tomls
-    }
-    yield test_scenarios
-
-
-def test_add_obs_points(test_db, test_scenarios):
-    test_scenario = test_scenarios["current_extreme12ft_no_measures.toml"]
-
-    test_scenario.init_object_model()
-    path_in = (
-        test_db.static_path
-        / "templates"
-        / test_scenario.site_info.attrs.sfincs.overland_model
+def default_sfincs_adapter(test_db) -> SfincsAdapter:
+    overland_path = test_db.static_path / "templates" / "overland"
+    with SfincsAdapter(model_root=overland_path) as adapter:
+        start_time = datetime(2023, 1, 1, 0, 0, 0)
+        duration = timedelta(hours=3)
+
+        adapter.set_timing(
+            TimeModel(start_time=start_time, end_time=start_time + duration)
+        )
+        adapter.logger = mock.Mock()
+        adapter.logger.handlers = []
+        adapter.logger.warning = mock.Mock()
+
+        # make sure model is as expected
+        # ? is it correct that the template model has waterlevels?
```

Comment:
is it?

Reply:
I added this comment this week.
@GundulaW